src/database.ts

```diff
@@ -1989,12 +1997,27 @@ class Database extends common.GrpcServiceObject {
    * });
    * ```
    */
-  getTransaction(): Promise<[Transaction]>;
-  getTransaction(callback: GetTransactionCallback): void;
+  getTransaction(callback: GetTransactionCallback): Promise<[Transaction]>;
+  getTransaction(
+    options: GetTransactionOptions,
+    callback: GetTransactionCallback
+  ): void;
   getTransaction(
-    callback?: GetTransactionCallback
+    optionsOrRunFn: GetTransactionOptions | GetTransactionCallback,
+    fn?: GetTransactionCallback
   ): void | Promise<[Transaction]> {
+    const callback =
+      typeof optionsOrRunFn === 'function'
+        ? (optionsOrRunFn as RunTransactionCallback)
+        : fn;
+    const options =
+      typeof optionsOrRunFn === 'object' && optionsOrRunFn
+        ? (optionsOrRunFn as RunTransactionOptions)
+        : {};
     this.pool_.getSession((err, session, transaction) => {
+      if (options.optimisticLock) {
+        transaction!.useOptimisticLock();
```

Comment:
Can we add a test that verifies that this option is actually sent to Spanner? From the PR description, this seems to be the most important part of the PR.

Reply:
Done